packages/components/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import * as RadixCheckbox from "@radix-ui/react-checkbox";
+import { CheckIcon } from "@radix-ui/react-icons";
+import { styled } from "../../stitches.config";
+
+const StyledBox = styled(RadixCheckbox.Root, {
+  backgroundColor: "transparent",
+  borderWidth: 1,
+  borderColor: "transparent",
+  padding: 0,
+  width: 24,
+  height: 24,
+  display: "inline-flex",
+  alignItems: "center",
+  justifyContent: "center",
+  defaultVariants: {
+    checkDisplay: "unchecked",
+  },
+  variants: {
+    disabledClick: {
+      true: {
+        backgroundColor: "#FFEED9",
+      },
+    },
+    checkDisplay: {
+      checked: {
+        backgroundColor: "$forest",
+        borderColor: "$forest",
+      },
+      unchecked: {
+        borderColor: "$wheat",
+        borderStyle: "solid",
+      },
+    },
+  },
+});
+
+const Check = styled(RadixCheckbox.Indicator, {
+  width: 14,
+  height: 14,
+  backgroundColor: "$forest",
+  color: "$sand",
+});
+
+export interface CheckboxProps
+  extends Omit<RadixCheckbox.CheckboxProps, "css"> {}
+const Checkbox = (props: CheckboxProps) => {
+  const [checked, setChecked] = useState(props.checked ?? false);
+  return (
+    <StyledBox
+      {...props}
+      disabledClick={props.disabled}
+      checkDisplay={checked ? "checked" : "unchecked"}
```

Comment:
why not just use the boolean?

Reply:
discovered boolean variants after i did this 😄 

I also greatly simplified this to not require variants and instead use attribute selectors (`data-state`) provided by radix-ui

Reply:
hell yea 🔥